security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -335,6 +324,62 @@ func (sc *SecretCache) GenerateSecret(ctx context.Context, connectionID, resourc
 	return ns, nil
 }
 
+func (sc *SecretCache) addFileWatcher(file string, token string, connKey ConnKey) {
+	// TODO(ramaraochavali): add integration test for file watcher functionality.
+	// Check if this file is being already watched, if so ignore it. FileWatcher has the functionality of
+	// checking for duplicates. This check is needed here to avoid processing duplicate events for the same file.
+	sc.certMutex.Lock()
+	npath := filepath.Clean(file)
+	_, exists := sc.fileCerts[npath]
+	if !exists {
+		if _, exists := sc.fileCerts[npath]; !exists {
+			sc.fileCerts[npath] = make(map[ConnKey]struct{})
+		}
+		sc.fileCerts[npath][connKey] = struct{}{}
+	}
```

Comment:
Do you need to check and add the connKey to sc.fileCerts[npath] when the npath exists?